.github/workflows/main.yml

```diff
@@ -0,0 +1,44 @@
+name: CI
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+      - cibuildwheel_based_CI
+        # TODO Remove line above and uncomment line below
+        #  workflow_dispatch:
```

Comment:
We can walk through it in the meeting, but the gist is that you have two commit trees (the current state of `main` and this branch). As this branch got merged from the branch that was used for PR #3, it still has all of that commit history in it, so if we just rebase this branch on top of `main`

```
git rebase origin/main
```

it will fail because we're asking Git to _replay_ all the commits in order on top of the current state of `main`, but `main` already has the first commits in this branch in it.

So what we want to do instead is to use an interactive Git rebase to rebase on top of the last commit that was in the branch used in PR #3 (where this PR diverged). That happens to be f6760e1 so in the rebase you would:
* Start and interactive rebase on top of `origin/main` (`git rebase -i origin/main`)
* Drop all commit f6760e1 and all commits before it
* Save this and continue the rebase
* Then do a `git push --force-with-lease` to update this branch.

Here's an `asciinema` recording of what that looks like

https://asciinema.org/a/661507

but we can step through it together for your first time. :+1:

Reply:
Reference video on `git rebase`: https://youtu.be/hv8dhOEzQcM?si=mQeE9lWR9mxorbLT